### ISSUE TITLE
Connection Pooling

### ISSUE THREAD

**Issue by vitruvvius** [jtv/libpqxx#444]
Most interfaces from other languages (JDBC, by example) have connection pooling embedded by default - ie, coupling execution framework w/ handling pool of connections to database.

What do we think about providing a configurable pool of connections and embedding that by default into the library? As of today, every project must reinvent pooling

**Comment by KayEss** [comment]
There are several projects that add connection pooling for postgres.

If you need connection pooling you're generally much better off connecting to one of those and using its features. This has the advantage that the infrastructure for the pools can be managed in a well known way rather than rely on a million different ways from a million different apps.

From an operational stand point per-application connection pools tend to cause more problems than they solve given the existence of well known external pools.

**Comment by jtv** [comment]
I made a start on this once, actually, as a separate add-on project.  Right now I think async support is still a little limited, both from the point of view of supporting compilers and from the point of view of ability to wait asynchronously for connections to complete etc.  I'd also need a lot more C++ async expertise.  So if I started on this today, I don't think we'd get something that would stand the test of time.

So as things stand, I agree with @KayEss that this is a job for a separate package.  Let people with the right insights and knowledge compete.  And if they need new libpqxx features to do the job right, obviously I'm open to that.
